internal/commands/app/create.go

```diff
@@ -101,70 +104,66 @@ func (cmd *CommandCreate) Handler(profile *cli.Profile, ui terminal.UI, clients
-	// )
-	// if exportErr != nil {
-	// 	return exportErr
-	// }
-	// if idx := strings.LastIndex(zipPkgName, "_"); idx != -1 {
-	// 	zipPkgName = zipPkgName[:idx]
-	// }
-	// zipPkgPath := filepath.Join(dir, zipPkgName) // Is this the correct path
-	// if writeErr := local.WriteZip(zipPkgPath, zipPkg); writeErr != nil {
-	// 	return writeErr
-	// }
+	rowCount := 3
+	if cmd.inputs.DataSource != "" {
+		dsCluster, err := cmd.inputs.resolveDataSource(clients.Realm, groupID, newApp.ID)
+		if err != nil {
+			return err
+		}
+		serviceName := newApp.Name + "_cluster"
+		serviceConfig := map[string]interface{}{
+			"name": serviceName,
+			"type": "mongodb-atlas",
+			"config": map[string]interface{}{
+				"clusterName":         dsCluster,
+				"readPreference":      "primary",
+				"wireProtocolEnabled": false,
+			},
+		}
+		var dsPath string
+		switch loadedApp.ConfigVersion() {
+		case realm.AppConfigVersion20210101:
+			dsPath = fmt.Sprintf("data_sources/%s/config.json", serviceName)
+		case
+			realm.AppConfigVersion20200603,
+			realm.AppConfigVersion20180301:
+			dsPath = fmt.Sprintf("services/%s/config.json", serviceName)
+		default:
+			return errors.New("unsupported config version")
+		}
+		data, err := json.MarshalIndent(serviceConfig, local.ExportedJSONPrefix, local.ExportedJSONIndent)
+		if err != nil {
+			return err
+		}
+		err = local.WriteFile(filepath.Join(dir, dsPath), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+		err = loadedApp.Load()
+		if err != nil {
+			return err
+		}
+		rowCount++
+	}
 
 	if err := clients.Realm.Import(
 		newApp.GroupID,
 		newApp.ID,
-		loadedApp,
+		loadedApp.AppData,
 	); err != nil {
 		return err
 	}
 
 	headers := []string{"Info", "Details"}
-	rows := []map[string]interface{}{
-		{
-			"Info":    "Client App ID",
-			"Details": newApp.ClientAppID,
-		},
-		{
-			"Info":    "Realm Directory",
-			"Details": dir,
-		},
-		{
-			"Info":    "Realm UI",
-			"Details": fmt.Sprintf("%s/groups/%s/apps/%s/dashboard", profile.RealmBaseURL(), newApp.GroupID, newApp.ID),
-		},
-		{
-			"Info":    "Check out your app",
-			"Details": fmt.Sprintf("cd ./%s && realm-cli app describe", newApp.Name),
-		},
+	rows := make([]map[string]interface{}, 0, rowCount)
+	rows = append(rows, map[string]interface{}{"Info": "Client App ID", "Details": newApp.ClientAppID})
+	rows = append(rows, map[string]interface{}{"Info": "Realm Directory", "Details": dir})
+	rows = append(rows, map[string]interface{}{"Info": "Realm UI", "Details": fmt.Sprintf("%s/groups/%s/apps/%s/dashboard", profile.RealmBaseURL(), newApp.GroupID, newApp.ID)})
+	if cmd.inputs.DataSource != "" {
+		rows = append(rows, map[string]interface{}{"Info": "Data Source", "Details": cmd.inputs.DataSource})
 	}
 
 	ui.Print(terminal.NewTableLog("Successfully created app", headers, rows...))
+	ui.Print(terminal.NewDebugLog("Check out your app: cd ./%s && realm-cli app describe", cmd.inputs.Directory))
```

Comment:
lets create `terminal.NewFollowupLog` which takes in `message string` and `items ...interface{}`, and will otherwise replace what we have for `NewReferenceLinksLog`/`NewSuggestedCommandsLog`.  You could then also use that here